optimum/intel/neural_compressor/modeling_base_seq2seq.py

```diff
@@ -0,0 +1,419 @@
+                with a different name.
+            decoder_file_name(`str`, *optional*):
+                The decoder model file name. Overwrites the default file name and allows one to save the decoder model
+                with a different name.
+            decoder_with_past_file_name(`str`, *optional*):
+                The decoder with past key values model file name overwriting the default file name, allowing to save
+                the decoder model with a different name.
+        """
+        if self.config.torchscript:
+            torch.jit.save(self.encoder_model, os.path.join(save_directory, encoder_file_name))
+            torch.jit.save(self.decoder_model, os.path.join(save_directory, decoder_file_name))
+            if self.decoder_with_past_model is not None:
+                torch.jit.save(self.decoder_with_past_model, os.path.join(save_directory, decoder_with_past_file_name))
+        elif isinstance(self.encoder, IPEXModel):
+            self.encoder_model._model.save(os.path.join(save_directory, encoder_file_name))
+            self.decoder_model._model.save(os.path.join(save_directory, decoder_file_name))
+        elif isinstance(self.encoder, PyTorchModel):
+            encoder_state_dict = self.encoder_model._model.state_dict()
+            decoder_state_dict = self.decoder_model._model.state_dict()
+
+            if hasattr(self.encoder, "q_config"):
+                encoder_state_dict["best_configure"] = self.encoder_model.q_config
+            torch.save(encoder_state_dict, os.path.join(save_directory, encoder_file_name))
+            if hasattr(self.decoder, "q_config"):
+                decoder_state_dict["best_configure"] = self.decoder_model.q_config
+            torch.save(decoder_state_dict, os.path.join(save_directory, decoder_file_name))
+        else:
+            encoder_state_dict = self.encoder_model.state_dict()
+            decoder_state_dict = self.decoder_model.state_dict()
+
+            torch.save(encoder_state_dict, os.path.join(save_directory, encoder_file_name))
+            torch.save(decoder_state_dict, os.path.join(save_directory, decoder_file_name))
+        logger.info(f"Model weights saved to {save_directory}")
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        encoder_file_name: Optional[str] = ENCODER_NAME,
+        decoder_file_name: Optional[str] = DECODER_NAME,
+        decoder_with_past_file_name: Optional[str] = DECODER_WITH_PAST_NAME,
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        **kwargs,
+    ):
+        """
+        Loads a model and its configuration file from a directory or the HF Hub.
```

Comment:
I would remove the docstring from `_from_pretrained` (which is copy pasted everywhere so we can keep them in "public" method (`from_pretrained` or  `from_transformers`) which will be accessible in the documentation, it's also true for openvino, I will take care of removing it as it's not very readable)